external/llvm-project/mlir/test/Dialect/GPU/ops.mlir

```diff
@@ -251,6 +251,26 @@ module attributes {gpu.container_module} {
 
       gpu.return
     }
+
+    // CHECK-LABEL: gpu.func @mfma_i8_4xi32
+    // CHECK:       gpu.mfma(%{{.*}}, %{{.*}}, %{{.*}}) : i32, vector<4xi32> 
+    // CHECK-NEXT:  gpu.mfma(%{{.*}}, %{{.*}}, %{{.*}}) : i32, vector<4xi32>
+    gpu.func @mfma_i8_4xi32(%a : i32, %b : i32, %c : vector<4xi32>) {
```

Comment:
Does it make sense to test for failure to lower? I suppose it does, so that we'll know to update the tests if we expose those intrinsics

Reply:
I'm not sure there's a good way to test "unexposed lowering patterns" because then the unit test run result will be non-zero and it will be misunderstood as a real failure.

I think we are good in terms that `LowerGpuToROCDLOps.cpp` will immediately return a failure in lowering if this is a pattern not explicitly handled. If we ended up need to handle other type of i8 xdlops, it will be apparent at that point.

Here's what I see when I added the non-reduction version of unit test:

> 
within split at /root/llvm-project-mlir/external/llvm-project/mlir/test/Conversion/GPUToROCDL/gpu-to-rocdl.mlir:390 offset :17:21: error: failed to legalize operation 'gpu.mfma' that was explicitly marked illegal
    %result16xi32 = gpu.mfma(%arg_i32, %arg_i32, %arg_vi32x32) {instr = "mfma_i32_32x32x4i8"} : i32, vector<32xi32>
Failed Tests (1):
  MLIR :: Conversion/GPUToROCDL/gpu-to-rocdl.mlir